docs/ProcessingPipeline.md

```diff
@@ -0,0 +1,62 @@
+# Processing Pipeline
+
+Some applications might require that traces be altered or filtered out before they are sent upstream. The processing pipeline allows users to create *processors* to define such behavior.
+
+Processors can be any object that responds to `#call` accepting `trace` as an argument (which is an `Array` of `Datadog::Span`s.)
+
+For example:
+
+```ruby
+lambda_processor = ->(trace) do
+  # Processing logic...
+  trace
+end
+
+class MyCustomProcessor
+  def call(trace)
+    # Processing logic...
+    trace
+  end
+end
+custom_processor = MyFancyProcessor.new
+```
+
+`#call` blocks of processors *must* return the `trace` object; this return value will be passed to the next processor in the pipeline.
```

Comment:
Minor: "`#call` blocks" is a bit confusing, since it's mixing method and block/proc/lambda nomenclature. Perhaps "The `#call` method of such processors..."